src/wovnio/html/HtmlConverter.php

```diff
@@ -74,6 +82,50 @@ private function insertSnippet($dom) {
 
     $token = $this->token;
     $insert_tags = array('head', 'body', 'html');
+    foreach($insert_tags as $tag_name) {
+      $parents = $dom->find($tag_name);
+      if (count($parents) > 0) {
+        $parent = $parents[0];
+        $parent->innertext = "<script src=\"//j.wovn.io/1\" data-wovnio=\"key=$token\" data-wovnio-type=\"backend_without_api\" async></script>" . $parent->innertext;
+        return;
+      }
+    }
+  }
+
+  /**
+   * Insert hreflang tags for all supported_langs
+   *
+   * @param simple_html_dom_node $dom
+   */
+  // <link rel="alternate" hreflang="ja" href="https://wovn.io:/ja/">
+  public function insertHreflangTags($dom) {
+    $lang_codes = $this->store->convertToCustomLangCodeArray($this->store->settings['supported_langs']);
+    foreach ($dom->find('link') as $node) {
+      $hreflang = $node->getAttribute('hreflang');
+      if (in_array($hreflang, $lang_codes)) {
+        // remove node
+        $node->outertext = '';
+      }
+    }
+
+    $insert_tags = array('head', 'body', 'html');
+    foreach($insert_tags as $tag_name) {
+      $parents = $dom->find($tag_name);
+      if (count($parents) > 0) {
+        $parent = $parents[0];
+        $hreflangTags = array('');
+        foreach ($lang_codes as $lang_code) {
+          $href = Url::addLangCode($this->headers->url, $this->store, $lang_code, $this->headers);
+          array_push($hreflangTags, '<link rel="alternate" hreflang="' . $lang_code . '" href="' . $href . '">');
```

Comment:
`$lang_code` needs to follow ISO 639 specification in `hreflang` attribute:
```
array_push($hreflangTags, '<link rel="alternate" hreflang="' . Lang.iso639_1Normalization($lang_code) . '" href="' . $href . '">');
```

Reply:
Thanks, we also need to make sure that we are using any custom lang aliases defined. I'll fix this.

Reply:
yes, that's right

Reply:
Updated, please check